system/Model.php

```diff
@@ -796,37 +811,7 @@ public function update($id = null, $data = null): bool
      */
     protected function objectToRawArray($data, bool $onlyChanged = true, bool $recursive = false): ?array
     {
-        $properties = parent::objectToRawArray($data, $onlyChanged);
-
-        $primaryKey = null;
-
-        if ($data instanceof Entity) {
-            $cast = $data->cast();
-
-            // Disable Entity casting, because raw primary key data is needed for database.
-            $data->cast(false);
-
-            $primaryKey = $data->{$this->primaryKey};
-
-            // Restore Entity casting setting.
-            $data->cast($cast);
-        }
-
-        // Always grab the primary key otherwise updates will fail.
-        if (
-            // @TODO Should use `$data instanceof Entity`.
-            method_exists($data, 'toRawArray')
-            && (
-                ! empty($properties)
-                && ! empty($this->primaryKey)
-                && ! in_array($this->primaryKey, $properties, true)
-                && ! empty($primaryKey)
-            )
-        ) {
-            $properties[$this->primaryKey] = $primaryKey;
-        }
-
-        return $properties;
+        return parent::objectToRawArray($data, $onlyChanged);
```

Comment:
Do we need this at all? Can't we just call the parent?

Reply:
> However, it seems that RDB cannot handle cases where Entity has Entity anyway.

I guess some people were making drivers for NoSQL databases... however, I can't imagine sharing the same layer for both of them. But this is a different story.

Let's keep it as it is.

Reply:
> The goal with this change is to have BaseModel which does not contain any database related code so that it can easily be extended to support other kinds of databases like NoSQL document based
See #3930

Reply:
Yes, I know - honestly, I never liked this idea.